### PR TITLE
Include PHP Min Version code

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#116281](http://central.tri.be/issues/116281)
---
📹 http://i.bordoni.me/1bf0b5baf223

**Before updating addons**
screen shot 2019-01-17 at 01 16 37 am

**After all plugins require correct version**
screen shot 2019-01-17 at 01 18 21 am

https://github.com/moderntribe/the-events-calendar/pull/2381 https://github.com/moderntribe/event-tickets/pull/1064 https://github.com/moderntribe/events-community-tickets/pull/131 https://github.com/moderntribe/events-community/pull/369 https://github.com/moderntribe/event-tickets-plus/pull/682 https://github.com/moderntribe/events-pro/pull/1109 https://github.com/moderntribe/events-eventbrite/pull/278 https://github.com/moderntribe/events-filterbar/pull/160

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:57](https://github.com/moderntribe/the-events-calendar/blob/be797389975df38cc28fea5352e782ffad352480/src/functions/php-min-version.php#L57)

> via jenkins/codesniffer-phpcs

File: src/functions/php-min-version.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * @return string
+ */
+function tribe_events_not_php_version_message() {
+	return wp_kses_post( sprintf(
+			__( '<b>The Events Calendar</b> requires PHP %1$s or higher, and the plugin has now disabled itself.', 'the-events-calendar' ),
+			tribe_events_get_php_min_version()
+		) ) .
+		'<br />' .
+		esc_attr__( 'To allow better control over dates, advanced security improvements and performance gain.', 'the-events-calendar' ) .
```
Comment (sc0ttkclark) [inline src/functions/php-min-version.php]:
```suggestion
		esc_html__( 'To allow better control over dates, advanced security improvements and performance gain.', 'the-events-calendar' ) .
```

File: src/functions/php-min-version.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		'<br />' .
+		esc_attr( sprintf(
```
Comment (sc0ttkclark) [inline src/functions/php-min-version.php]:
```suggestion
		esc_html( sprintf(
```

File: src/functions/php-min-version.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			__( 'Contact your Hosting or your system administrator and ask for this Upgrade to version %1$s of PHP.', 'the-events-calendar' ),
+			tribe_events_get_php_min_version()
+		) );
+}
+
+/**
+ * Echos out the error for the PHP min version as a WordPress admin Notice
```
Comment (juanfra) [inline src/functions/php-min-version.php]: Echoes

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:57](https://github.com/moderntribe/the-events-calendar/blob/b1fa03cbb389f204fd9587693a19cdbffff300f3/src/functions/php-min-version.php#L57)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:57](https://github.com/moderntribe/the-events-calendar/blob/32095bafd7536e0ef2af7ae4920c3b7dcbafca1f/src/functions/php-min-version.php#L57)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:103](https://github.com/moderntribe/the-events-calendar/blob/1db218b70b6564bdea7be41bfa15141e6ec89703/src/functions/php-min-version.php#L103)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:103](https://github.com/moderntribe/the-events-calendar/blob/2b76ef652de35a71cade86885f5a8360d0a20c9b/src/functions/php-min-version.php#L103)

> via jenkins/codesniffer-phpcs

File: src/functions/php-min-version.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+		$label_names = current( $names );
+	} elseif ( 2 === $count_names ) {
+		$label_names = current( $names ) . $last_connector . end( $names );
+	} else {
+		$last_name = array_pop( $names );
+		$label_names = implode( $many_connector, $names ) . $last_connector . $last_name;
+	}
+
+	return wp_kses_post( sprintf(
+			__( '<b>%1$s</b> requires <b>PHP %2$s</b> or higher.', 'the-events-calendar' ),
```
Comment (Camwyn) [inline src/functions/php-min-version.php]: Should this include a descriptive comment for translators that tells them what the variables are? I know we don't translate the plugin names or version numbers, but...
https://codex.wordpress.org/I18n_for_WordPress_Developers#Descriptions

File: src/functions/php-min-version.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ * Echoes out the error for the PHP min version as a WordPress admin Notice
+ *
+ * @since  TBD
+ *
+ * @return void
+ */
+function tribe_not_php_version_notice() {
+	echo '<div id="message" class="error"><p>' . tribe_not_php_version_message() . '</p></div>';
```
Comment (Camwyn) [inline src/functions/php-min-version.php]: escape?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:103](https://github.com/moderntribe/the-events-calendar/blob/a63f96471ac9f39104be1ed867ec94b69798d92c/src/functions/php-min-version.php#L103)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [php-min-version.php:103](https://github.com/moderntribe/the-events-calendar/blob/74913135bb65a223f4b025053cefe63a13d4dcce/src/functions/php-min-version.php#L103)

> via jenkins/codesniffer-phpcs

**Pull request merged**